### DISCUSSION TITLE
error on postal web interface - We're sorry, but something went wrong.

### DISCUSSION THREAD

**Discussion by davids355** [postalserver/postal#2813] [Help with using Postal] [answered]
Since upgrading to the latest version of postal I keep seeing this error in the dashboard - 
We're sorry, but something went wrong.
If you are the application owner check the logs for more information.

I also have users complaining that emails are not being sent.
When I restart postal it starts working again but seems to fail shortly after.

I found these threads suggesting that the issue might be related to either database of messages stuck in the queue - 
https://github.com/postalserver/postal/issues/981
https://github.com/postalserver/postal/issues/521

However, I am unable to find the files being referenced above.

Can someone please advise me how to find and edit these files (or if there is another potential cause/fix for the problem)?

Thanks in advance.

**Comment by willpower232** [comment]
If you try `postal logs web` you can see what the web error is and whatever is affecting that will be causing all the issues from the sounds of things.

**Comment by davids355** [comment]
Thanks @willpower232 

Snippet from log - 
It looks like there is a mysql problem? Not sure how to further troubleshoot or fix?

postal-web-1  | [8] [2024-02-22T21:07:19.359] INFO -- : Completed 500 Internal Server Error in 17ms (ActiveRecord: 5.6ms | Allocations: 2468)
postal-web-1  | [8] [2024-02-22T21:07:19.360] FATAL -- :
postal-web-1  | **ActionView::Template::Error (MySQL client is not connected):**
postal-web-1  |     26:             = link_to [organization, server], :class => 'largeList__link' do
postal-web-1  |     27:               %span.largeList__rightLabel.label{:class => "label--serverStatus-#{server.status.underscore}"}= t("server_statuses.#{server.status.underscore}")
postal-web-1  |     28:               %p= server.name
postal-web-1  |     29:               %p.largeList__subText #{number_with_precision server.message_rate, :precision => 2} messages/minute
postal-web-1  |     30:       .js-searchable__empty.is-hidden
postal-web-1  |     31:         .noData.noData--clean
postal-web-1  |     32:           %p.noData__title No servers were found...

**Reply by willpower232** [reply]
Potentially, how are you hosting your databases? Are they dockerised on the same host as Postal? Do you have enough memory available?

**Comment by davids355** [comment]
The databases are dockerised on the same host as postal yes. There is enough ram - according to top command - 7752mb total, 4402 free, 1571 used.

**Reply by willpower232** [reply]
It might be worth double checking your MySQL settings and see if they differ greatly when compared to https://github.com/postalserver/postal/issues/324#issuecomment-323538575

**Comment by davids355** [comment]
I have updated SQL settings according to that thread, however it has not helped. Interestingly, I mentioned that after I restart postal (Or the entire server) it starts working. But upon further checks, it still does not fully work - after restart I can get into the web interface, I can go into messages > outgoing messages and also messages > send message.

But if I go into messages > queue, it crashes with the same error - 

We're sorry, but something went wrong.
If you are the application owner check the logs for more information.

Also, just for the record this is a very low volume setup - typically 25-50 emails per day and perhaps once a month there may be a mailing campaign going out which would be 1000 emails (Just mentioning in case this is relevant).

**Reply by willpower232** [reply]
The error messages are hidden by default so they all present in the UI with the same message.

You can click around every other page apart from the queue? And even after the queue page shows the error you can get to the other pages?

**Comment by davids355** [comment]
Yes; even after I visit the queue and the error presents itself, the rest of the pages continue to work. Queue is the only one I’ve noticed that won’t work, but I’ve not tested all pages.

also, after a certain amount of time, it appears to stop working completely; at this point none of the pages work, until the server is restarted

**Reply by willpower232** [reply]
this sounds like two separate errors, are you able to check the logs after you visit the queue page a few times to see what that one is?

**Comment by davids355** [comment]
Thanks for the replies Will especially on a Sunday I appreciate it. I will check logs again in relation to that error, but probably will have to do it later this evening. I’ll post here once I have the details.

**Comment by davids355** [comment]
I tried again loading queue page and got the following entries in the log file - see screenshot below for reference.

<img width="937" alt="logs" src="https://github.com/postalserver/postal/assets/149956199/930eac77-1fa2-4f94-9992-475d302d3451">

**Reply by willpower232** [reply, marked as answer]
so there is a message in the queue which has been stuck there so long that the original message has been cleared out by postal

thankfully it shows you the id 66326 so I suggest you connect to the database manually and see if you can delete the message from queued messages, I expect there will be some date values in that row which confirm it is quite old and irrelevant at this point so safe to delete

alternatively, you could include my fix from #521 as it sounds about right here

**Comment by davids355** [comment]
Also, after checking, it seems that queue page is the only one producing that error currently.

**Comment by davids355** [comment]
I have checked the database and I could not find any entry with ID matching 66326 in the messages table. However, I did find that ID number in Links table. I dont know if thats relevant though?

As for the fix in 521, I am not sure how to impliment that could you help me? If I ipostal/app/controllers/servers_controller.rb but I cannot find that folder or file. Assuming I should be looking in /opt/postal/ but in that directory I only have caddy-data, config, environments, install and monitor_certs.sh

**Reply by willpower232** [reply]
if its definitely the right database then the link may be related.

The file is in the docker container so you would have to edit the file within the container or overwrite it with a docker volume or make your own docker image with the change in.

**Comment by davids355** [comment]
Thanks Will. I am not sure that I have the capability to do that. Do you know of any freelancers that provide paid support for postal? Or would it be sufficient to find someone that understands docker? I can do the basics but this is getting a bit beyond me :-)

**Comment by davids355** [comment]
OK, I have spent some more time this morning troubleshooting based on your instructions.
I tried to impliment the update as per #521 but I could not get it to work. This is what I was trying - 

I copied contents of this file - 
https://github.com/postalserver/postal/blob/main/app/controllers/servers_controller.rb

And placed it on my server here - 
/opt/postal/app/controllers/servers_controller.rb

I edited it as per notes in the above thread, to include new code in the queue definition.

I then tried editing the docker compose file to include an additional volume as follows - 
/opt/postal/app/controllers:/opt/postal/app/app/controllers

Then restarted postal, but when I went into the docker container and looked at the file it did not have the changes that I had made.

Where was I going wrong with this? And also, which docker container should I have been adding that code to?
(I would just like to know this so that I can better understand how it is supposed to work).

Having said all of that, I gave up on the above and decided to look again at the database like you had suggested previously and when looking I found there were two databases - one called postal, which doesnt have much in it at all (So I had originally thought it was not relevant) and one called postal-server-2 (Which is quite large).

I was looking originally at the large database but when I went back and checked the smaller one, I found two messages in the queue! one had the ID mentioned in the logs.

I deleted it with the following commands - 
mariadb
use postal;
DELETE FROM queued_messages WHERE message_id=66326;

After that it still was not working but when I checked the logs, the other message was also coming up as causing a problem, so I deleted that one as well, now the queue page is loading successfully!!

thanks for your help @willpower232
